src/Angor/Client/Pages/Create.razor

```diff
@@ -1181,19 +1168,30 @@
             return false;
         }
 
-        var prev = DateTime.UtcNow;
-        foreach (var stage in project.ProjectInfo.Stages)
+        if (project.ProjectInfo.Stages.Any())
         {
-            if ((stage.ReleaseDate - prev).Days < 0)
+            // First check if first stage is after project start
+            if (project.ProjectInfo.Stages.First().ReleaseDate < project.ProjectInfo.StartDate)
             {
-                notificationComponent.ShowErrorMessage("Stages must be chronological");
+                notificationComponent.ShowErrorMessage("First stage must be after project start date");
                 return false;
             }
 
-            prev = stage.ReleaseDate;
+            // Then check stages are in order
+            var sortedStages = project.ProjectInfo.Stages.OrderBy(s => s.ReleaseDate).ToList();
+            for (int i = 0; i < sortedStages.Count - 1; i++)
+            {
+                if (sortedStages[i].ReleaseDate >= sortedStages[i + 1].ReleaseDate)
+                {
+                    notificationComponent.ShowErrorMessage($"Stage {i + 2} must be after stage {i + 1}");
+                    return false;
+                }
+            }
+
+            // Keep stages sorted
+            project.ProjectInfo.Stages = sortedStages;
```

Comment:
I don't like this so much as we are making change on behalf of the user, we should give a warning and let the user fix it.